### ISSUE TITLE
[Accessibility] [Global] [ButtonGroup] Orientation> Buttons are not programmatically grouped in a semantic 'list'

### ISSUE THREAD

**Issue by silvalaura** [cengage/react-magma#1999]
[A11Y-512](https://jira.cengage.com/browse/A11Y-512)

**Environment:**
Windows 11/Chrome 137/JAWS 2025
Windows 11/Edge 136/NVDA 2025
Mac 15/Safari/VoiceOver

**Steps to Reproduce:**

1. Open the URL – https://react-magma.cengage.com/version/4.9.1/api/tooltip/
2. Turn on the screen reader (JAWS/NVDA/VoiceOver).
3. Navigate to the 'ButtonGroup' component page from the left navigation.
4. Under the 'Orientation' section, locate the group of buttons labeled ‘1’, ‘2’, ‘3’, ‘4’, ‘5’, and ‘6’.
5. Navigate through these buttons using the Tab key or arrow keys and observe how screen reader announces them.

**Actual Result:**
The buttons are visually presented in a vertical group, but they are not programmatically grouped using semantic list markup. Screen reader users receive no indication that the buttons are part of a related items as it does not read list markup such as "list of 6 items", "list item 1 of 6", "list item 2 of 6" etc.

Refer to screencast:
https://app.screencast.com/UpnQZytRsbB0P

**Note:** This issue is reproducible for all of the Button Groups present on the page.

**Expected Result:**
All of the buttons within the group should be programmatically tagged in a 'list' for screen reader users.

**User Impact:**
Without semantic list markup, screen reader users would be unable to identify the total no. of buttons available in the group and their relative position within the group.

WCAG Guideline:
1.3.1 Info and Relationships (A)

**Bug Fix Recommendation:**
Wrap the button group in a semantic `<ul>` element with each button inside an `<li>`.

**Comment by olehnoskov-cengage** [comment]
**Tested on:**
- V4:
Docs: https://next--upbeat-sinoussi-f675aa.netlify.app/api/buttongroup/
Storybook: https://storybook-preview-v4-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/buttongroup--default

- V5: 
Docs: https://rc--upbeat-sinoussi-f675aa.netlify.app/api/buttongroup/
Storybook: https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/buttongroup--default

Everything works as expected. No bugs found.

**Comment by silvalaura** [comment]
This causes errors in unit tests, therefore it was reopened. We cherry picked the commit out of the last release.

**Comment by olehnoskov-cengage** [comment]
**Tested on:**
- V4:
Docs: https://next--upbeat-sinoussi-f675aa.netlify.app/api/buttongroup/
Storybook: https://storybook-preview-v4-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/buttongroup--default

- V5: 
Docs: https://rc--upbeat-sinoussi-f675aa.netlify.app/api/buttongroup/
Storybook: https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/buttongroup--default

Everything works as expected. No bugs found.
